app/collections/settings.js

```diff
@@ -1,89 +1,91 @@
-import {
-    saveSuccessful,
-    saveFailed,
-    unableToRetrieve,
-    updateFailed
-} from '../notifications/general';
-
-const Datastore = require('nedb');
-const electron = require('electron');
-const path = require('path');
-
-const userDataPath = (electron.app || electron.remote.app).getPath('userData');
-const collectionsPath = path.join(userDataPath, 'collections');
-const Settings = new Datastore({
-    filename:              path.join(collectionsPath, 'settings.db'),
-    autoload:              true,
-    corruptAlertThreshold: 1,
-    timestampData:         true
-});
-
-export const saveGradeSystem = data => {
-    Settings.insert(data, error => {
-        if (error) {
-            updateFailed();
-        }
-    });
-};
-
-export const getAddressData = () => new Promise(resolve => Settings.find({}, (err, entry) => {
-    if (err) {
-        unableToRetrieve();
+import connectionToDB from './connectionSetup';
+import {displayToast} from '../notifications';
+
+const Settings = connectionToDB('settings');
+
+export async function saveGradeSystem(data) {
+    try {
+        await Settings.insert(data);
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
     }
+}
+
+export async function getAddressData() {
+    try {
+        const results = await Settings.find({});
 
-    return resolve(entry);
-}));
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
 
-export const getSystemType = () => new Promise(resolve => Settings.find({}, (err, docs) => {
-    if (err) {
-        updateFailed();
+        return null;
     }
+}
 
-    return resolve(docs);
-}));
+export async function getSystemType() {
+    try {
+        const results = await Settings.find({});
+
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
+
+        return null;
+    }
+}
 
-const updateAddress = (previous, id) => {
+async function updateAddress(previous, id) {
     const {title, street, province, country, zip, city, year} = previous;
-    Settings.update(
-        {_id: id},
-        {$set: {title, street, province, country, zip, city, year}},
-        {},
-        err => {
-            if (err) {
-                saveFailed();
-            }
-            saveSuccessful();
-        }
-    );
-};
-
-const updateSystem = (previous, id) => {
+    try {
+        await Settings.update({_id: id}, {$set: {title, street, province, country, zip, city, year}}, {});
```

Comment:
@hcuffy you COULD (this is just about code style) name the parameter `_id` and the first object in the `update` function would just be `{_id}`